tests/utils/test_dist__gather__losses.py

```diff
@@ -0,0 +1,173 @@
+        x = batch[0]
+        y = batch[1]
+        x = self.model(x).flatten(start_dim=1)
+        y = y.flatten(start_dim=1)
+        loss: Tensor = self.criterion(x, y)
+        return loss
+
+    def configure_optimizers(self) -> Any:
+        return SGD(self.parameters(), lr=self.learning_rate)
+
+
+@pytest.fixture
+def close_torch_distributed() -> Generator[None, None, None]:
+    yield None
+    torch.distributed.destroy_process_group()
+
+
+class TestGatherLayer_Losses:
+    """
+    Tests that the gather layer works as expected.
+
+    Because using a DDPStrategy causes the whole script to be executed multiple
+    times, running a proper test is difficult. The approach used here:
+
+    1. This test was run once with n_devices=1 and gather=False. The resulting
+    parameters after 10 epochs are hardcoced as expected_params__10_epochs__no_gather.
+
+    2. This test is now run with n_devices=2 and gather=True. The resulting
+    parameters are asserted to be the same ad with n_devices=1 and gather=False.
+
+    Note that the results would not be the same for n_devices=1 and n_devices=2 if
+    there was:
+    - Any randomness in the transform, as the order the samples are processed and
+    thus the random seed differ.
+    - Any batch normalization in the model, as the batch size is split between
+    the devices and there is no gather layer for the batch normalization.
+    """
+
+    def test_loss_ntxent(self, close_torch_distributed: None) -> None:
+        self._test_ddp(
+            NTXentLoss,
+            torch.Tensor(
+                [
+                    [0.1675, 0.2676, 0.3678, 0.4679, 0.5680],
+                    [0.5763, 0.6742, 0.7721, 0.8700, 0.9679],
+                ]
+            ),
+            learning_rate=0.1,
+        )
+
+    def test_loss_tico(self, close_torch_distributed: None) -> None:
+        self._test_ddp(
+            TiCoLoss,
+            torch.Tensor(
+                [
+                    [0.1985, 0.3004, 0.4022, 0.5041, 0.6060],
+                    [0.5694, 0.6627, 0.7559, 0.8492, 0.9424],
+                ]
+            ),
+            learning_rate=0.1,
+        )
+
+    def test_loss_vicreg(self, close_torch_distributed: None) -> None:
+        self._test_ddp(
+            VICRegLoss,
+            torch.Tensor(
+                [
+                    [-0.2382, -0.1381, -0.0381, 0.0619, 0.1620],
+                    [-0.0494, 0.0505, 0.1504, 0.2503, 0.3503],
+                ]
+            ),
+            learning_rate=1e-8,
+        )
+
+    def test_loss_dcl(self, close_torch_distributed: None) -> None:
+        self._test_ddp(
+            DCLLoss,
+            torch.Tensor(
+                [
+                    [0.1484, 0.2335, 0.3185, 0.4036, 0.4886],
+                    [0.6198, 0.7093, 0.7988, 0.8884, 0.9779],
+                ]
+            ),
+            learning_rate=0.1,
+        )
+
+    def _test_ddp(
+        self,
+        loss: Module,
```

Comment:
```suggestion
        loss: Type[Module],
```

Not sure how that passed mypy